src/test/kotlin/se/uulm/snowballr/backend/service/user/UpdateTest.kt

```diff
@@ -0,0 +1,89 @@
+        val updateFieldMask = FieldMaskUtil.fromStringList(listOf("first_name", "last_name"))
+        val request = User.Update.newBuilder().setUser(updatedUser).setMask(updateFieldMask).build()
+
+        coEvery { userRepoMock.getUserById(any()) } returns nonAdminUser
+
+        assertDoesNotThrow { mainService.updateUser(request) }
+    }
+
+    @Test
+    fun `When an admin updates a user's information successfully, then no exception is thrown`() = testCoroutine {
+        val adminUser = DataBuilder.createExampleUser(role = UserRole.USER_ROLE_ADMIN)
+        val updatedUser = DataBuilder.createExampleUser(firstName = "John", lastName = "Doe").toGrpcUser()
+        val updateFieldMask = FieldMaskUtil.fromStringList(listOf("first_name", "last_name"))
+        val request = User.Update.newBuilder().setUser(updatedUser).setMask(updateFieldMask).build()
+
+        coEvery { userRepoMock.getUserById(any()) } returns adminUser
+
+        assertDoesNotThrow { mainService.updateUser(request) }
+    }
+
+    @Test
+    fun `When a user wants to change its email to an existing email, then an exception is thrown`() = testCoroutine {
+        val request = User.Update.newBuilder().setMask(FieldMaskUtil.fromString("email")).build()
+
+        coEvery { userRepoMock.doesUserExistByEmail(any()) } returns true
+
+        assertThrows<SnowballRException.DuplicateEntityException.UserEmail> { mainService.updateUser(request) }
+    }
+
+    @Test
+    fun `When a non-admin user wants to change the user role, then an exception is thrown`() = testCoroutine {
+        val nonAdminUser = DataBuilder.createExampleUser(role = UserRole.USER_ROLE_DEFAULT)
+        val request = User.Update.newBuilder().setMask(FieldMaskUtil.fromString("role")).build()
+
+        coEvery { userRepoMock.getUserById(any()) } returns nonAdminUser
+
+        assertThrows<SnowballRException.UnauthorizedException.Single.User> { mainService.updateUser(request) }
+    }
+
+    @Test
+    fun `When a non-admin user wants to change the user information from another user, then an exception is thrown`() =
+        testCoroutine {
+            val anotherUser = DataBuilder.createExampleUser()
+            val updatedUser = DataBuilder.createExampleUser().toGrpcUser()
+            val updateFieldMask = FieldMaskUtil.fromString("first_name")
+            val request = User.Update.newBuilder().setUser(updatedUser).setMask(updateFieldMask).build()
+
+            coEvery { userRepoMock.getUserById(any()) } returns anotherUser
+
+            assertThrows<SnowballRException.UnauthorizedException.Single.User> { mainService.updateUser(request) }
+        }
+
+    @Test
+    fun `When an error occurs while updating a user, then an exception is thrown`() = testCoroutine {
+        val request = User.Update.newBuilder().build()
+
+        coEvery { userRepoMock.updateUser(any()) } throws Exception("Failed to persist user")
+
+        assertThrows<Exception> { mainService.updateUser(request) }
```

Comment:
Use the `TestSpecificException` so that we can assure that the test case fails because of this mocked exception.
```suggestion
        coEvery { userRepoMock.updateUser(any()) } throws TestSpecificException()

        assertThrows<TestSpecificException> { mainService.updateUser(request) }
```

Reply:
This I new, right?

Reply:
yes